website/blog/2022/04/17/api-observability.md

```diff
@@ -0,0 +1,278 @@
+
+## Logs
+
+**Logs** are also easy to instrument and trivial steps of API observability, they can be used to inspect API calls in real-time for debugging, auditing, and recording time-stamped events that happened over time. There are several logger plugins Apache APISIX provides such as:
+
+* [http-logger](https://apisix.apache.org/docs/apisix/plugins/http-logger/)
+
+* [skywalking-logger](https://apisix.apache.org/docs/apisix/plugins/skywalking-logger/)
+
+* [tcp-logger](https://apisix.apache.org/docs/apisix/plugins/tcp-logger)
+
+* [kafka-logger](https://apisix.apache.org/docs/apisix/plugins/kafka-logger)
+
+* [rocketmq-logger](https://apisix.apache.org/docs/apisix/plugins/rocketmq-logger)
+
+* [udp-logger](https://apisix.apache.org/docs/apisix/plugins/udp-logger)
+
+* [clickhouse-logger](https://apisix.apache.org/docs/apisix/plugins/clickhouse-logger)
+
+* [error-logger](https://apisix.apache.org/docs/apisix/plugins/error-log-logger)
+
+* [google-cloud-logging](https://apisix.apache.org/docs/apisix/plugins/google-cloud-logging)
+
+And you can see the [full list](https://apisix.apache.org/docs/apisix/plugins/zipkin) on the official website of Apache APISIX. Now for demo purposes, let's choose a simple but mostly used _http-logger_ plugin that is capable of sending API Log data requests to HTTP/HTTPS servers or sends as JSON objects to Monitoring tools. We can assume that a route and an upstream are created. You can learn how to set up them in the **[Getting started with Apache APISIX](https://youtu.be/dUOjJkb61so)** video tutorial. Also, you can find all command-line examples on the GitHub page [apisix-observability-plugins](https://boburmirzo.github.io/apisix-observability-plugins/).
+
+You can generate a mock HTTP server at [mockbin.com](https://mockbin.org/) to record and view the logs. Note that we also bind the route to an upstream (You can refer to this documentation to learn about more [core concepts of Apache APISIX](https://apisix.apache.org/docs/apisix/architecture-design/apisix)).
+
+The following is an example of how to enable the http-logger for a specific route.
+
+```
+curl http://127.0.0.1:9080/apisix/admin/routes/1 \
+-H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
+{
+    "plugins": {
+          "http-logger": {
+              "uri": "http://mockbin.org/bin/5451b7cd-af27-41b8-8df1-282ffea13a61"
+          }
+     },
+    "upstream_id": "1",
+    "uri": "/get"
+}'
+
+```
+
+> To http-logger plugin settings, your can just put your mock server URI address like below:
+
+```
+    {"uri": "http://mockbin.org/bin/5451b7cd-af27-41b8-8df1-282ffea13a61"}
+Once we get a successful response from APISIX server, we can send a request to this _get_ endpoint to generate logs.
+curl -i http://127.0.0.1:9080/get
+  
+```
```

Comment:
The format is out of order. Line 97 is a single code, line 98 is the body, and line 99 is the command. Please adjust it again.